### PR TITLE
Added FlatButton component, adjusted themeType

### PR DESCRIPTION
**Pull request by woutsluiter**
This pr adds:
- a FlatButton component for text style buttons.
- nested ThemeType properties for readability.

### CODE REVIEW

File: src/components/Button/Button.style.tsx [hunk 1]
```diff
@@ -40,34 +48,10 @@ const Button:StyledComponentClass<PropsType, ThemeType> = styled(ButtonTemplate)
...
-                theme.buttonPrimaryBackgroundColor,
-                theme.buttonPrimaryBorderRadius,
-                theme.buttonPrimaryBorderWidth,
-                theme.buttonPrimaryBorderColor,
-                theme.buttonPrimaryColor,
-            );
+            case 'secondary': return applyTheme(theme.button.secondary);
+            case 'destructive': return applyTheme(theme.button.destructive);
+            case 'warning': return applyTheme(theme.button.warning);
+            default: return applyTheme(theme.button.primary);
```
Comment (CarloPalinckx) [inline src/components/Button/Button.style.tsx]: 💎

File: src/themes.ts [hunk 2]
```diff
@@ -31,27 +18,142 @@ type ThemeType = {
...
-    buttonSecondaryBorderColor: '#d2d7e0',
-    buttonSecondaryBorderRadius: '0',
-    buttonSecondaryBorderWidth: '1px',
-    buttonSecondaryColor: '#333740',
-    buttonWarningBackgroundColor: '#FCC200',
-    buttonWarningBorderColor: '#FCC200',
-    buttonWarningBorderRadius: '0',
-    buttonWarningBorderWidth: '1px',
-    buttonWarningColor: '#fff',
+    button: {
```
Comment (CarloPalinckx) [inline src/themes.ts]: I really like the nesting, however I think we should consider what the shape of the theme will look like. In other words: 
- What would define a nesting. 
- Would we make a "common" nesting containing things like `fontFamily` and `fontSize`. 
- Is there a way (if we even should) we could enforce boundaries on those nestings? (ie: button components can only use `theme.button` etc.).

Could you maybe create an issue and write up a little proposal so we can document this?

Comment (woutsluiter) [reply src/themes.ts]: Created issue https://github.com/MyOnlineStore/bricks/issues/20

Comment (CarloPalinckx) [reply src/themes.ts]: 👍

**Pull request merged**